UNITTESTS/fakes/ble/source/GattServerImpl_mock.cpp

```diff
@@ -0,0 +1,101 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2021 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "GattServerImpl_mock.h"
+#include "util/HijackMember.h"
+
+// Accessors to private callbacks of GattCharacteristic
+HIJACK_MEMBER(read_cb_access, FunctionPointerWithContext<GattReadAuthCallbackParams *> GattCharacteristic::*, &GattCharacteristic::readAuthorizationCallback);
+HIJACK_MEMBER(write_cb_access, FunctionPointerWithContext<GattWriteAuthCallbackParams *> GattCharacteristic::*, &GattCharacteristic::writeAuthorizationCallback);
```

Comment:
I think a nicer solution would be to make public getters for these, there's not risk in a getter